src/reddit/reddit_post.ts

```diff
@@ -103,12 +107,20 @@ export default class RedditPost {
   }
 
   /** Checks the title to determine if the post is an update. */
-  public hasValidTitle(titleFilter: RegExp): boolean {
+  public hasValidTitle(titleFilter?: RegExp): boolean {
+    if (!titleFilter) {
+      return true;
+    }
+
     return titleFilter.test(this.title);
   }
 
   /** Checks if the submission is new. */
-  public isNew(date: Date): boolean {
+  public isNew(date?: Date): boolean {
+    if (!date) {
+      return true;
+    }
```

Comment:
So when the submission doesn't have a date it's always considered to be new?

Reply:
Yes, hmm I guess we need to find a better solution

Reply:
Nvm, this is not the submission date, but the filter date, so this should be correct.